messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/JGroupBroadcastGroupAdd.java

```diff
@@ -0,0 +1,161 @@
+import static org.wildfly.extension.messaging.activemq.CommonAttributes.JGROUPS_CLUSTER;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.activemq.artemis.api.core.BroadcastEndpointFactory;
+import org.apache.activemq.artemis.api.core.BroadcastGroupConfiguration;
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.ServiceRegistry;
+import org.wildfly.clustering.dispatcher.CommandDispatcherFactory;
+import org.wildfly.extension.messaging.activemq.broadcast.CommandDispatcherBroadcastEndpointFactory;
+import org.wildfly.extension.messaging.activemq.logging.MessagingLogger;
+
+/**
+ * Handler for adding a broadcast group.
+ *
+ * @author Brian Stansberry (c) 2011 Red Hat Inc.
+ */
+public class JGroupBroadcastGroupAdd extends AbstractAddStepHandler {
+
+    public static final JGroupBroadcastGroupAdd INSTANCE = new JGroupBroadcastGroupAdd(true);
+    public static final JGroupBroadcastGroupAdd LEGACY_INSTANCE = new JGroupBroadcastGroupAdd(false);
+
+    private final boolean needLegacyCall;
+    private JGroupBroadcastGroupAdd(boolean needLegacyCall) {
+        super(JGroupBroadcastGroupDefinition.ATTRIBUTES);
+        this.needLegacyCall= needLegacyCall;
+    }
+
+    @Override
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+        CommonAttributes.renameChannelToCluster(operation);
+        if (operation.hasDefined(JGROUPS_CLUSTER.getName())) {
+            if (operation.hasDefined(JGROUPS_CHANNEL_FACTORY.getName()) && !operation.hasDefined(JGROUPS_CHANNEL.getName())) {
+                // Handle legacy behavior
+                String channel = operation.get(JGROUPS_CLUSTER.getName()).asString();
+                operation.get(JGROUPS_CHANNEL.getName()).set(channel);
+
+                PathAddress channelAddress = context.getCurrentAddress().getParent().getParent().getParent().append(ModelDescriptionConstants.SUBSYSTEM, "jgroups").append("channel", channel);
+                ModelNode addChannelOperation = Util.createAddOperation(channelAddress);
+                addChannelOperation.get("stack").set(operation.get(JGROUPS_CHANNEL_FACTORY.getName()));
+                // Fabricate a channel resource if it is missing
+                context.addStep(addChannelOperation, AddIfAbsentStepHandler.INSTANCE, OperationContext.Stage.MODEL);
+            }
+        }
```

Comment:
If we eliminate the deprecated JGROUPS_CHANNEL_FACTORY attribute from this new resource definition, we don't need any special handling here.  All of this logic can remain in the handler for the legacy resource.

Reply:
I kept it for compatibility reasons: the legacy operations don't really do something except register a placeholder resource. That's why i kept also the same attribute names to avoid translating them